src/admin/importers/ThemeImporter.php

```diff
@@ -0,0 +1,136 @@
+{
+    private $packageInstaller;
+
+    /**
+     * @inheritdoc
+     */
+    public function run()
+    {
+        $this->packageInstaller = Yii::$app->getPackageInstaller();
+
+        $exists = [];
+
+        foreach (Yii::$app->themeManager->getThemes(false) as $theme) {
+            /** @var \luya\theme\Theme $theme */
+            $exists = array_merge($exists, $this->handleThemeDefinitionInDirectories($theme->basePath));
+        }
+
+        foreach (Theme::find()->all() as $theme) {
+            if (!in_array($theme->id, $exists) && $theme->delete()) {
+                $this->addLog("[!] The theme {$theme->base_path} does not found anymore and was deleted.");
+            }
+        }
+
+        return $this->addLog("Theme importer finished with " . count($exists) . " themes.");
+    }
+
+    /**
+     * Handle an array with definitions whether they are files or folders.
+     *
+     * @param array $definitions
+     *
+     * @return array
+     */
+    protected function handleThemeDefinitions(array $definitions)
+    {
+        $ids = [];
+        foreach ($definitions as $themeDefinition) {
+            $ids = array_merge($ids, $this->handleThemeDefinitionInDirectories($themeDefinition));
+        }
+
+        return $ids;
+    }
+
+    /**
+     * Handle a theme definition for different folders
+     *
+     * @param string $themeDefinition
+     *
+     * @return array
+     */
+    protected function handleThemeDefinitionInDirectories($themeDefinition)
+    {
+        $results = [];
+
+        $themeDefinition = preg_replace('#^vendor/#', '@vendor/', $themeDefinition);
+
+        $themeId = $this->saveThemeByPath($themeDefinition);
+        if ($themeId) {
+            $results[$themeDefinition] = $themeId;
+        } else {
+            $this->addLog("Unable to find '{$themeDefinition}'");
+        }
+
+        return $results;
+    }
+
+    protected function saveThemeByPath($themeDefinition)
+    {
+        $path = Yii::getAlias($themeDefinition);
+        if (is_dir($path)) {
+            return $this->saveTheme($themeDefinition);
+        }
+
+        return null;
+    }
+
+    /**
+     * Save a theme by its base path.
+     * Example path: @app/themes/blank
+     *
+     * @param $basePath
+     *
+     * @return mixed|bool False if could not exists.
+     * @throws \yii\base\InvalidConfigException
+     */
+    protected function saveTheme($basePath)
+    {
+        $themeFile = Yii::getAlias($basePath . '/theme.json');
+        if (!file_exists($themeFile)) {
+            return false;
+        }
+
+        $config = Json::decode(file_get_contents($themeFile)) ?: [];
+
+        $themeConfig = new ThemeConfig($basePath, $config);
+
+        $themeModel = Theme::findOne(['base_path' => $basePath]);
+
+        if (!$themeModel) {
+            $themeModel = new Theme();
+            $themeModel->base_path = $basePath;
+            $themeModel->json_config = Json::encode($themeConfig->toArray());
```

Comment:
Maybe we should integrate the encoding part in the model:

```
public function init()
{
    parent::init();

    $this->on(self::EVENT_AFTER_VALIDATE, function() {
      $this->json_config = Json::encode($this->json_config);
   }
}
```

Reply:
I would prefer a function like `loadThemeConfig(ThemeConfig $config)`.